soh/src/code/z_parameter.c

```diff
@@ -1481,10 +1481,30 @@ void Inventory_SwapAgeEquipment(void) {
         }
 
         gSaveContext.adultEquips.equipment = gSaveContext.equips.equipment;
+        if (CVarGetInteger("gSwitchAge", 0) || CVarGetInteger("gSwitchTimeline", 0)) {
+            for (i = 0; i < ARRAY_COUNT(gSaveContext.equips.buttonItems); i++) {
+                gSaveContext.equips.buttonItems[i] = gSaveContext.childEquips.buttonItems[i];
+
+                if (i != 0) {
+                    gSaveContext.equips.cButtonSlots[i - 1] = gSaveContext.childEquips.cButtonSlots[i - 1];
+                }
+
+                if (((gSaveContext.equips.buttonItems[i] >= ITEM_BOTTLE) &&
+                     (gSaveContext.equips.buttonItems[i] <= ITEM_POE)) ||
+                    ((gSaveContext.equips.buttonItems[i] >= ITEM_WEIRD_EGG) &&
+                     (gSaveContext.equips.buttonItems[i] <= ITEM_CLAIM_CHECK))) {
+                    osSyncPrintf("Register_Item_Pt(%d)=%d\n", i, gSaveContext.equips.cButtonSlots[i - 1]);
+                    gSaveContext.equips.buttonItems[i] =
+                        gSaveContext.inventory.items[gSaveContext.equips.cButtonSlots[i - 1]];
+                }
+            }
```

Comment:
A comment explaining this part could be helpful as well, I'm not sure I understand why these are special-cased.

Reply:
My bad yea, should have made this more clear but also the diff makes it less easy to tell too. I basically separated the flow for vanilla and using the switch age variables when going from adult to child because it seemed more appropriate to do rather than a bunch of if statements surrounding certain sections with conditions doubling up. 

This section in particular basically just checks the current status of the bottle to see if the item was changed at all between the last time you changed age, and updates accordingly.

Reply:
Ah I see, and it looks like it does trade quest items as well?

Reply:
Yeah that too, anything which would be able to change value basically (i think ocarina and hookshot are exceptions due to being only obtainable before changing age/being only obtainable by the age it can be equipped)